### ISSUE TITLE
Callback was already called

### ISSUE THREAD

**Issue by falxcerebri** [wavded/winston-mail#33]
I wonder how this is happening again:
My npm ls looks like this for the part that's interesting (my dependency tree is quite long)
```
...
└─┬ winston-mail@1.0.1
  ├─┬ emailjs@1.0.0
  │ ├── addressparser@0.3.2
  │ ├── bufferjs@1.1.0
  │ ├─┬ mimelib@0.2.14
  │ │ ├── addressparser@0.2.1
  │ │ └─┬ encoding@0.1.12
  │ │   └── iconv-lite@0.4.13
  │ ├── moment@1.7.0
  │ └── starttls@0.2.1
  └── underscore@1.8.3
```
What I get is:
```
[...]\node_modules\winston\node_modules\async\lib\async.js:38
            if (called) throw new Error("Callback was already called.");
                        ^

Error: Callback was already called.
    at [...]\node_modules\winston\node_modules\async\lib\async.js:38:31
    at [...]\node_modules\winston\lib\winston\logger.js:236:7
    at Object.callback ([...]\node_modules\winston-mail\lib\winston-mail.js:110:5)
    at begin ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\client.js:54:25)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
    at initiate ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:412:11)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
    at response ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:383:9)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
    at response ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:221:11)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
    at response ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:199:9)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
    at Object.SMTP.send ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:190:7)
    at Object.SMTP.command ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:207:10)
    at Object.SMTP.helo ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:228:10)
    at attempt ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:387:23)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
    at response ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:309:11)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
    at response ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:199:9)
    at caller ([...]\node_modules\winston-mail\node_modules\emailjs\smtp\smtp.js:49:14)
 [nodemon] app crashed - waiting for file changes before starting...
```
I've just replaced my dir and app name with [...] .
What is funny, that my email is being sent successfully. Has anyone experienced this? I'm using Win10 and running Avast, Node v4.2.4.

**Comment by falxcerebri** [comment]
With my workaround (simplistic try/catch) - it works for me. The callback is executed once and the app doesn't die.

**Comment by falxcerebri** [comment]
#34

**Comment by falxcerebri** [comment]
I've created a convenience pull request for fixing this: #34

**Comment by erktime** [comment]
Not sure if anyone else has had this issue, but for me it was the result of having a bad mail server configuration. I needed to enable ```options.tls``` rather than ```options.secure```

**Comment by vikaskanani** [comment]
I am using `option.tls` but still face this issue.

**Comment by endquote** [comment]
I ran into this too. Trying @dgilling's fix.

**Comment by falxcerebri** [comment]
Yeah, maybe my pull request will get some attention. It's exactly what @dgilling posted later, adding `console.log(...)`

**Comment by wavded** [comment]
This is unfortunate.  I'm at a loss as to why the callback would be called more than once.  Seems perhaps something upstream in Winston itself changed at some point that I'm not familiar with.

**Comment by wavded** [comment]
Released in `1.3.0`
